src/net/discovery/config.rs

```diff
@@ -0,0 +1,204 @@
+/// query parameters
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct Origin {
+    scheme: Scheme,
+    authority: Authority,
+}
+
+impl From<Origin> for Uri {
+    fn from(origin: Origin) -> Self {
+        Uri::builder()
+            .scheme(origin.scheme)
+            .authority(origin.authority)
+            .path_and_query("")
+            .build()
+            .unwrap()
+    }
+}
+
+impl Serialize for Origin {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&format!("{}://{}", self.scheme, self.authority))
+    }
+}
+
+impl<'de> Deserialize<'de> for Origin {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct UrlVisitor;
+        impl<'de> serde::de::Visitor<'de> for UrlVisitor {
+            type Value = Origin;
+
+            fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+                formatter.write_str("a valid format origin made up of http[s]://<authority>")
+            }
+
+            fn visit_str<E: serde::de::Error>(self, v: &str) -> Result<Self::Value, E> {
+                let uri = v.parse::<Uri>().map_err(|err| E::custom(err.to_string()))?;
+                let parts = uri.into_parts();
+                let scheme = parts.scheme.ok_or_else(|| E::custom("missing scheme"))?;
+                let authority = parts
+                    .authority
+                    .ok_or_else(|| E::custom("missing authority"))?;
+                Ok(Origin { scheme, authority })
+            }
+        }
+        deserializer.deserialize_str(UrlVisitor)
+    }
+}
+
+/// Configuration values relevant for interacting with other helpers
+#[derive(Serialize, Deserialize)]
+struct Peer {
+    origin: Origin,
+    public_key: PublicKey,
+}
+
+/// All config value necessary to discover other peer helpers of the MPC ring
+#[derive(Serialize, Deserialize)]
+struct Config {
+    peers: [Peer; 3],
+}
+
+impl Config {
+    /// Reads config from `file_location`. Expects file to be json format
+    /// # Errors
+    /// if the file does not exist, or is in an invalid format
+    #[allow(dead_code)] // TODO: will use in upcoming PR
+    pub fn new(file_location: PathBuf) -> Result<Self, Error> {
+        use std::{fs::File, io::BufReader};
+
+        let contents = File::open(file_location)?;
+        let buffered = BufReader::new(contents);
+        Ok(serde_json::from_reader(buffered)?)
+    }
+}
+
+impl PeerDiscovery for Config {
+    /// TODO: support HTTPS
+    fn peers(&self) -> [MpcHelperClient; 3] {
+        self.peers
+            .iter()
+            .map(|peer| MpcHelperClient::new(peer.origin.clone().into()))
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap_or_else(|_| panic!("could not convert vec into array"))
+    }
+
+    // TODO: do we need different public key for HTTP and PRSS, or can they share?
```

Comment:
I thought about it again and I think the answer is no. Security is one issue, but also TLS may use RSA with public keys being 10x larger than ECC. Martin knows better whether we should use RSA or ECC to encrypt the traffic (for some reason RSA is very popular and I don't have a good explanation why it is preferred over ECC), but there are technical limitations on public key crypto as well. So imo we should use different public keys

Reply:
We should have a different key for the TLS server side of things, definitely.  TLS and PRSS are two separate purposes and using the same key for both isn't possible anyway.

We'll want to use ECC, but probably ECDSA with P-256.  RSA signing is maybe OK, but it's unnecessarily inefficient.